app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,67 @@ def show
     @post = Post.find(params[:id])
     @user = User.find(params[:user_id])
   end
+
+  def new
+    @post = Post.new
+    @user = current_user
+    @id = params[:user_id]
+  end
+
+  def create
+    @post = Post.create(post_params)
+    @user = current_user
+    @post.author_id = @user.id
+
+    # respond_to block
+    respond_to do |f|
+      f.html do
+        if @post.save
+          flash[:success] = 'Post created successfully!'
+          redirect_to "/users/#{@user.id}/posts/"
+        else
+          flash.now[:error] = 'Error creating post!'
+          render :new
+        end
+      end
+    end
+  end
+
+  # Create comment
+  def comment_post
+    # new object from params
+    @comment = Comment.new(comment_params)
+    @comment.author = current_user
+    @comment.post = Post.find(params[:id])
+
+    if @comment.save
+      flash[:success] = 'Comment created successfully!'
+      redirect_to user_post_path
+    else
+      flash.now[:error] = 'Error creating comment!'
+      render :new
+    end
+  end
```

Comment:
- [ ] I can see that your implementations are working fine 👍 
However, picture a scenario where your app scales and this `post_controller` where you have included both `comment` and `likes` functionalities increases in logic and other features. It will be difficult to locate and debug errors or even manage right? 
 
One solution to that problem is to let each controller manage its logic and other features.  That's why the previous reviewer made some suggestions that you create `comments_controller` and `likes_controller`.

I want to suggest you please spare some time to work on it. You are almost there.